delete SequentialTests, replace with SequencedTests

marcrasi opened the pull request:
SequentialTests is hard to maintain and debug, as discussed in https://github.com/tensorflow/swift-apis/pull/683. We already have tests covering most of the things that it covers. The only thing we don't have tests for is `sequenced`. Therefore, this PR deletes SequentialTests and adds some tests covering `sequenced`.

cc @t-ae

t-ae commented on Tests/TensorFlowTests/SequencedTests.swift.gyb:
```diff
@@ -0,0 +1,42 @@
+        let input = Float(0)
+	let output = input.sequenced(
+	    through: ${', '.join(['AddOne()' for _ in range(count)])})
+	XCTAssertEqual(output, ${count})
```
Indentation is broken.
```suggestion
    let input = Float(0)
    let output = input.sequenced(
        through: ${', '.join(['AddOne()' for _ in range(count)])})
    XCTAssertEqual(output, ${count})
```

dan-zheng commented:
All Swift code in [tensorflow/swift-apis](https://github.com/tensorflow/swift-apis) was automatically formatted via [`swift-format`](https://github.com/apple/swift-format) in https://github.com/tensorflow/swift-apis/pull/711.

[Formatting configuration here](https://github.com/tensorflow/swift-apis/blob/master/.swift-format).
One notable change is that we're now using 2-space indentation, which is consistent with:
- [Google Swift Style Guide](https://google.github.io/swift)
- The Swift standard library style
- The style of [apple/swift-format](https://github.com/apple/swift-format)

---

As a result, this pull request now has merge conflicts due to code formatting changes. Please resolve the conflicts if you'd like us to re-review your patch! If you haven't already, please consider installing [`swift-format`](https://github.com/apple/swift-format) so you can auto-format code by yourself in the future!

marcrasi commented:
> I think there is still some issues with the indentation?

Fixed.

> Also, since this is supposed to be differentiable, could we have a test that checks the differentiability and ensures the gradients are good?

Done!

> Also, I haven't checked, but can you confirm that we have a test that covers the function builder Sequential implementation somewhere?

I found one but it's non-orthogonal and it only tests one of the aritys. I added a gybbed one that tests all the aritys.
